rudder-core/src/main/scala/com/normation/rudder/repository/jdbc/ExpectedReportsJdbcRepository.scala

```diff
@@ -193,31 +212,19 @@ class FindExpectedReportsJdbcRepository(
             select N.nodejoinkey, N.nodeid, N.nodeconfigids, generate_subscripts(N.nodeconfigids,1) as v
             from expectedreportsnodes N
           ) as NN
-          where ${inT2(("NN.nodeid","NN.nodeconfigids[v]"), nodeConfigIds)}
+          where ${in("NN.nodeconfigids[v]", nodeConfigIds.map(_.version.value))}
         ) as NNN
         on E.nodejoinkey = NNN.nodejoinkey
       """ + rulePredicate
 
-      for {
-        entries <- tryo ( jdbcTemplate.query(query, ReportAndNodeMapper).asScala )
-      } yield {
-        toExpectedReports(entries).toSet
-      }
-    }
-  }
-  // returns all the pending: those where the expected don't contain the actual
-  def getPendingReports(previousAndExpected: Set[PreviousAndExpectedNodeConfigId], filterByRules: Set[RuleId]) : Box[Set[RuleExpectedReports]] = {
-    if(previousAndExpected.isEmpty) Full(Set())
-    else {
-      val expectedNodeConfigIds = previousAndExpected.map( x => (x.expected))
-      val previousNodeConfigIds = previousAndExpected.map( x => (x.previous))
-
-      val rulePredicate = if(filterByRules.isEmpty) "" else " where " + in("ruleid", filterByRules.map(_.value))
 
-       val query = s"""select
-            E.pkid, E.nodejoinkey, E.ruleid, E.directiveid, E.serial, E.component, E.componentsvalues
-          , E.unexpandedComponentsValues, E.cardinality, E.begindate, E.enddate
-          , NNN.nodeid, NNN.nodeconfigids
+      /**
+       * Trying to reduce the number of returned rows by the query.
+       * But it does make so little difference on the quantity of data transfered
+       * that it does not matter.
+       */
+      val query_subscript_configids_3 = s"""select X.pkid, array_agg(array_to_string(X.nodeid || X.nodeconfigids, ',')) as nodeconfigs from (
```

Comment:
here you kept unuased sql query; is that voluntary ? 
It make the code harder to read, since the name of the strings are very similar